kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp

```diff
@@ -339,5 +339,109 @@ KRATOS_TEST_CASE_IN_SUITE(ReplaceElementsAndConditionsProcess3, KratosCoreFastSu
+
+    std::vector<NodeType::Pointer> element_nodes_1 (3);
+    element_nodes_1[0] = p_node_1;
+    element_nodes_1[1] = p_node_3;
+    element_nodes_1[2] = p_node_4;
+    Triangle2D3 <NodeType>::Pointer p_triangle_1 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_1} );
+
+    std::vector<NodeType::Pointer> element_nodes_2 (3);
+    element_nodes_2[0] = p_node_2;
+    element_nodes_2[1] = p_node_5;
+    element_nodes_2[2] = p_node_3;
+    Triangle2D3 <NodeType>::Pointer p_triangle_2 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_2} );
+
+    std::vector<NodeType::Pointer> element_nodes_3 (3);
+    element_nodes_3[0] = p_node_5;
+    element_nodes_3[1] = p_node_6;
+    element_nodes_3[2] = p_node_3;
+    Triangle2D3 <NodeType>::Pointer p_triangle_3 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_3} );
+
+    // Elements are created
+    Element::Pointer p_elem_0 = Kratos::make_intrusive<Element>(1, p_triangle_0, p_elem_prop);
+    Element::Pointer p_elem_1 = Kratos::make_intrusive<Element>(2, p_triangle_1, p_elem_prop);
+    Element::Pointer p_elem_2 = Kratos::make_intrusive<Element>(3, p_triangle_2, p_elem_prop);
+    Element::Pointer p_elem_3 = Kratos::make_intrusive<Element>(4, p_triangle_3, p_elem_prop);
+
+    // Element with id 4 is present in this model part and should therefore not be replaced
+    parent_model_part.AddElement(p_elem_0);
+    parent_model_part.AddElement(p_elem_1);
+    parent_model_part.AddElement(p_elem_2);
+    parent_model_part.AddElement(p_elem_3);
+
+    this_model_part.AddElement(p_elem_0);
+    this_model_part.AddElement(p_elem_1);
+    this_model_part.AddElement(p_elem_2);
+
+    sister_model_part.AddElement(p_elem_0);
+    sister_model_part.AddElement(p_elem_1);
+    sister_model_part.AddElement(p_elem_3);
+
+    // Compute process
+    Parameters settings( R"(
+    {
+        "element_name":"Element2D3N",
+        "condition_name": "Condition2D2N"
+    }  )" );
+
+    ReplaceElementsAndConditionsProcess process(this_model_part, settings);
+    process.Execute();
+
+    std::vector<int> modified_elems_ids = {1 ,2 ,3};
+    std::string component_name;
+    for (auto& r_element : this_model_part.Elements()) {
+        CompareElementsAndConditionsUtility::GetRegisteredName(r_element, component_name);
+        KRATOS_CHECK_EQUAL(component_name, "Element2D3N");
+    }
+
+    for (auto& r_element : sister_model_part.Elements()) {
+        if (std::find(modified_elems_ids.begin(), modified_elems_ids.end(), r_element.Id()) != modified_elems_ids.end()) {
+            CompareElementsAndConditionsUtility::GetRegisteredName(r_element, component_name);
+            KRATOS_CHECK_EQUAL(component_name, "Element2D3N");
+        }
+        else {
+            KRATOS_CHECK_EXCEPTION_IS_THROWN(CompareElementsAndConditionsUtility::GetRegisteredName(r_element, component_name),"Element \"N6Kratos7ElementE\" not found!");
```

Comment:
hm this is a strange check but I know you have to do it bcs there is only one type of element in the core
Not a blocker for me, maybe @pooyan-dadvand has a better suggestion

I thought this was machine specific but since Travis accepts it I guess it is ok

Reply:
@pooyan do you think we should change this or do you have any other solution?

Reply:
I would avoid this check as it relies on name mangling and can be changed due to compiler or compiling options.